src/celeritas/em/distribution/WentzelDistribution.hh

```diff
@@ -0,0 +1,345 @@
+    // Target isotope
+    IsotopeView const& target_;
+
+    // Mott coefficients for the target element
+    WentzelElementData const& element_data_;
+
+    //! Calculates the form factor from the given form coefficient and
+    //! scattered polar angle
+    inline CELER_FUNCTION real_type calculate_form_factor(real_type form_coeff,
+                                                          real_type cos_t) const;
+    //! Helper function for calculating the flat form factor
+    inline CELER_FUNCTION real_type flat_form_factor(real_type x) const;
+
+    //! Calculate incident momentum squared
+    inline CELER_FUNCTION real_type inc_mom_sq() const;
+
+    //! Calculate the nuclear form momentum scale
+    inline CELER_FUNCTION real_type nuclear_form_momentum_scale() const;
+
+    //! Calculate the screening coefficient R^2 for electrons
+    inline CELER_FUNCTION real_type screen_r_sq_elec() const;
+};
+
+//---------------------------------------------------------------------------//
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with state and data from WentzelInteractor
+ */
+CELER_FUNCTION
+WentzelDistribution::WentzelDistribution(real_type inc_energy,
+                                         real_type inc_mass,
+                                         IsotopeView const& target,
+                                         WentzelElementData const& element_data,
+                                         real_type cutoff_energy,
+                                         bool is_electron,
+                                         WentzelRef const& data)
+    : data_(data)
+    , inc_energy_(inc_energy)
+    , inc_mass_(inc_mass)
+    , is_electron_(is_electron)
+    , cutoff_energy_(cutoff_energy)
+    , target_(target)
+    , element_data_(element_data)
+{
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Samples the final direction of the interaction. This direction is in
+ * the frame where the incident particle's momentum is oriented along the
+ * z-axis, so it's final direction in the lab frame will need to be rotated.
+ */
+template<class Engine>
+CELER_FUNCTION Real3 WentzelDistribution::operator()(Engine& rng) const
+{
+    UniformRealDistribution<real_type> uniform_sample;
+
+    real_type screen_coeff = compute_screening_coefficient();
+    real_type cos_t_max_elec = compute_max_electron_cos_t();
+
+    // Parameters for scattering of a nucleus
+    real_type form_factor_coeff = 0;
+    real_type cos_t1 = 1;
+    real_type cos_t2 = -1;
+
+    // Randomly choose if scattered off of electrons instead
+    WentzelXsCalculator xsec(
+        target_.atomic_number(), screen_coeff, cos_t_max_elec);
+    real_type elec_ratio = xsec();
+    if (uniform_sample(rng) < elec_ratio)
+    {
+        // TODO: Can simplify this logic with
+        //       -1 <= cos_t_max_elec <= 1
+        cos_t1 = max(cos_t1, cos_t_max_elec);
```

Comment:
For `0 <= t, t_max <= pi`, if `t < t_max` then `cos t > cos t_max` so we indeed want to choose the maximum of the cosines, which would be the minimums of the angles. I can probably go ahead and apply the simplification since I only left it out for testing purposes.